### PR TITLE
TF-625: unit test

### PR DESCRIPTION
**Pull request by ManhNTX**
Create unit test for search use case. @hoangdat @dab246  Please check it

### CODE REVIEW

**Comment by hoangdat** [comment]
Please more detail on commit's message

**Comment by hoangdat** [comment]
```
00:23 +33 -3: /home/runner/work/jmap-dart-client/jmap-dart-client/test/jmap/email/search/use_case_test/search_emails_inbox_use_case_test.dart: search email test Search emails in Inbox which have the string report [E]
  DioError [DioErrorType.other]: Assertion failed: "Could not find mocked route matching request for POST /jmap { data: {methodCalls: [[Email/query, {accountId: 0eacc7a5c74b27ab36a823bc5c34da36e16c093705f241d6ed5f48ee73a4ecfb, filter: {inMailbox: c2543650-32cf-11eb-995c-a3ae66e9f96a, text: report}, limit: 3}, c2], [Email/get, {accountId: 0eacc7a5c74b27ab36a823bc5c34da36e16c093705f241d6ed5f48ee73a4ecfb, #ids: {resultOf: c2, name: Email/query, path: ids/*}, properties: [id, subject, size, from, receivedAt, sentAt, preview, hasAttachment]}, c3]], using: [urn:ietf:params:jmap:core, urn:ietf:params:jmap:mail]}, query parameters: {}, headers: {accept: application/json;jmapVersion=rfc-8621, content-type: application/json; charset=utf-8, content-length: 526} }"
```

File: test/jmap/email/search/use_case_test/search_by_form_use_case_test.dart [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      sentAt: UTCDate(DateTime.parse("2022-06-10T04:44:03Z")),
+      receivedAt: UTCDate(DateTime.parse("2022-06-10T04:51:41Z")));
+
+  Future<List<Email>?> searchMailByCondition(Comparator comparator) async {
+    final baseOption = BaseOptions(method: 'POST');
+    final dio = Dio(baseOption)..options.baseUrl = 'http://domain.com';
+    final dioAdapter = DioAdapter(dio: dio);
+    dioAdapter.onPost(
+        '/jmap',
+            (server) => server.reply(200, {
```
Comment (dab246) [inline test/jmap/email/search/use_case_test/search_by_form_use_case_test.dart:39]: The parameters of a function should align with each other. 
```
'/jmap'
(server) => server.reply(200, {})
```

File: test/jmap/email/search/use_case_test/search_by_after_use_case_test.dart [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      ..addFilters(
+        EmailFilterCondition(
+          after: UTCDate(
+            DateTime.parse('2022-05-12T06:12:00Z'),
+          ),
+          from: 'manh',
+        ),
+      );
+    final queryEmailInvocation = jmapRequestBuilder.invocation(queryEmailMethod,
+        methodCallId: MethodCallId('c1'));
```
Comment (dab246) [inline test/jmap/email/search/use_case_test/search_by_after_use_case_test.dart:144]: IMO there's no need to pass `methodCallId`, you should let it auto-generate. If I remember correctly, it starts with `c0`

File: test/jmap/email/search/use_case_test/search_by_form_use_case_test.dart [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        },
+        headers: {
+          "accept": "application/json;jmapVersion=rfc-8621",
+          "content-type": "application/json; charset=utf-8"
+        });
+
+    final httpClient = HttpClient(dio);
+    final processingInvocation = ProcessingInvocation();
+    final jmapRequestBuilder =
+    JmapRequestBuilder(httpClient, processingInvocation);
```
Comment (dab246) [inline test/jmap/email/search/use_case_test/search_by_form_use_case_test.dart:129]: Should write `final jmapRequestBuilder = JmapRequestBuilder(httpClient, processingInvocation);`

**Review by hoangdat — APPROVED** [review]

**Review by dab246 — COMMENTED** [review]
IMO, You should write some failed cases for search.

File: test/jmap/email/search/use_case_test/search_by_after_use_case_test.dart (hunk 2 above)
Comment (hoangdat) [reply test/jmap/email/search/use_case_test/search_by_after_use_case_test.dart:144]: `methodCallId` is optional, `processingInvocation` handle to make it unique. But no problem if you want to use it yourself :)

Comment (ManhNTX) [reply test/jmap/email/search/use_case_test/search_by_after_use_case_test.dart:144]: @dab246 I think if I add `methodCallId`, method request will be clearer

**Review by hoangdat — APPROVED** [review]

**Review by dab246 — APPROVED** [review]

**Pull request merged**